packages/tangle/schedulerutils/accessmanacache.go

```diff
@@ -0,0 +1,64 @@
+package schedulerutils
+
+import (
+	"time"
+
+	"github.com/iotaledger/hive.go/identity"
+)
+
+const (
+	// manaCacheLifeTime time after which the mana vector should be refreshed.
+	manaCacheLifeTime = 5 * time.Second
+)
+
+// AccessManaCache is a structure which provides access to cached access mana values.
+// Mana values are refreshed after certain amount of time has passed.
+type AccessManaCache struct {
+	rawAccessManaVector        map[identity.ID]float64
+	updatedAccessManaVector    map[identity.ID]float64
+	cacheRefreshTime           time.Time
+	minMana                    float64
+	accessManaMapRetrieverFunc func() map[identity.ID]float64
+}
+
+// NewAccessManaCache returns a new AccessManaCache.
+func NewAccessManaCache(accessManaMapRetrieverFunc func() map[identity.ID]float64, minMana float64) *AccessManaCache {
+	accessManaCache := &AccessManaCache{
+		minMana:                    minMana,
+		accessManaMapRetrieverFunc: accessManaMapRetrieverFunc,
+	}
+	accessManaCache.refreshCacheIfNecessary()
+	return accessManaCache
+}
+
+// GetCachedMana returns cached access mana value for a given node and refreshes mana vectors if they expired
+// currently returns at least MinMana.
+func (a *AccessManaCache) GetCachedMana(id identity.ID) float64 {
+	a.refreshCacheIfNecessary()
+
+	if mana, ok := a.updatedAccessManaVector[id]; ok && mana >= a.minMana {
+		return mana
+	}
+	// TODO: return actual mana value instead of MinMana
+	// always return at least MinMana
+	a.updatedAccessManaVector[id] = a.minMana
+	return a.updatedAccessManaVector[id]
+}
```

Comment:
I assume we are returning minMana to get 0 mana messages to be scheduled?

I don't like the idea that `getCachedMana` returns a fake mana. You explained it in the comments, but it is still somewhat misleading. I would have a wrapper function like `roundToMinMana` or maybe add a non-optional boolean parameter like `roundToMin` in the function just to stick it to people's face that we're doing this funky stuff.

 I also don't understand why you are changing the underlying vector instead of simply doing `return a.minMana`
 I just wouldn't expect a getter to do that. Also imagine if we find another use for access mana (I noticed we like changing stuff) and then reuse this cache that has this unexpected behavior of overridin the real mana value.
 
 But maybe if we do that TODO soon all of what I said doesn't really matter :-P (just change it to ISSUE:XXX)